src/file_tree.ml

```diff
@@ -134,7 +153,7 @@ let is_temp_file fn =
   || String.is_suffix fn ~suffix:".swp"
   || String.is_suffix fn ~suffix:"~"
 
-let load ?(warn_when_seeing_jbuild_file=true) path =
+let load ~warn_when_seeing_jbuild_file path =
   let rec walk path ~dirs_visited ~project ~data_only : Dir.t =
     let contents = lazy (
```

Comment:
I reviewed this function for for side effects, and discovered this `lazy`, which is normally thought of as a "benign" side-effect.
Unfortunately, in combination with `Memo` (as we just realized) lazy is no longer benign.

Consider code like this:
````
let g = memo blah in
let f = memo (fun x -> lazy (g x)) in
let a = memo (fun x -> force (f x)) in
let b = memo (fun x -> force (f x)) in
````
here, only one of `a x` and `b x` will ever pick up a dependency on `g`.
So we need to forbid lazy in combination with memo and introduce our own (e.g. `Memo.lazy`).